### PR TITLE
[IMP] Updating state sms as sent when saved

### PR DESCRIPTION
**Pull request by Kev-Roche**
Change sms state as "sent" when they are sent and saved. Otherwise, the sms state keeps its default value : outgoing.

### CODE REVIEW

**Comment by Kev-Roche** [comment]
cc @sebastienbeau , @bguillot

File: sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py [hunk 1]
```diff
@@ -5,30 +5,50 @@
...
+        self.assertEqual(self.sms2.state, "outgoing")
+
+        with self.mockSMSGateway():
+            self.sms2.send()
+        self.assertEqual(self.sms2.state, "sent")
+
+    def test_sms_unlink(self):
+        # force_unlink=False by default
+        self.sms1.unlink()
+        self.assertEqual(len(self.sms_sms.search([])), 3)
```
Comment (sebastienbeau) [inline sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py]: do not do a search, as this test will have random result.
If a module create a sms and so you already have an sms the qty will be 4.

you can simply do

```
self.sms1.unlink()
self.assertTrue(self.sms1.exists())
self.sms1.with_context(force_unlink=True).unlink()
self.assertFalse(self.sms1.exists())
```

Comment (Kev-Roche) [reply sms_no_automatic_delete/tests/test_purge_and_unlink_sms.py]: Indeed, it's better your way, I did the modification

**Review by sebastienbeau — APPROVED** [review]
LGTM (code review) please squash

**Comment by Kev-Roche** [comment]
cc @sebastienbeau

**Comment by sebastienbeau** [comment]
@Kev-Roche I think you did something wrong we should see only one commit here

**Review by PierrickBrun — APPROVED** [review]
LGTM

**Pull request merged**